utils/rpms/daos.spec

```diff
@@ -0,0 +1,227 @@
+Requires: fuse >= 3.4.2
+Requires: protobuf-c
+Requires: spdk
+Requires: fio < 3.4
+
+%description
+The Distributed Asynchronous Object Storage (DAOS) is an open-source
+software-defined object store designed from the ground up for
+massively distributed Non Volatile Memory (NVM). DAOS takes advantage
+of next generation NVM technology like Storage Class Memory (SCM) and
+NVM express (NVMe) while presenting a key-value storage interface and
+providing features such as transactional non-blocking I/O, advanced
+data protection with self healing on top of commodity hardware, end-
+to-end data integrity, fine grained data control and elastic storage
+to optimize performance and cost.
+
+%package server
+Summary: The DAOS server
+Requires: %{name} = %{version}-%{release}
+Requires(post): /sbin/ldconfig
+Requires(postun): /sbin/ldconfig
+
+%description server
+This is the package needed to run a DAOS server
+
+%package client
+Summary: The DAOS client
+Requires: %{name} = %{version}-%{release}
+
+%description client
+This is the package needed to run a DAOS client
+
+%package tests
+Summary: The DAOS test suite
+Requires: %{name}-client = %{version}-%{release}
+Requires: python-pathlib
+
+%description tests
+This is the package needed to run the DAOS test suite
+
+%package devel
+Summary: The DAOS development libraries and headers
+
+%description devel
+This is the package needed to build software with the DAOS library.
+
+%prep
+%setup -q
+%setup -q -a 1
+
+
+%build
+# remove rpathing from the build
+rpath_files="utils/daos_build.py"
+rpath_files+=" $(find . -name SConscript)"
+sed -i -e '/AppendUnique(RPATH=.*)/d' $rpath_files
+
+scons %{?no_smp_mflags}    \
+      --config=force       \
+      USE_INSTALLED=all    \
+      PREFIX=%{?buildroot}
+
+%install
+scons %{?no_smp_mflags}              \
+      --config=force                 \
+      install                        \
+      USE_INSTALLED=all              \
+      PREFIX=%{?buildroot}%{_prefix}
+BUILDROOT="%{?buildroot}"
+PREFIX="%{?_prefix}"
+sed -i -e s/${BUILDROOT//\//\\/}[^\"]\*/${PREFIX//\//\\/}/g %{?buildroot}%{_prefix}/TESTING/.build_vars.*
+mv %{?buildroot}%{_prefix}/lib{,64}
+#mv %{?buildroot}/{usr/,}etc
+mkdir -p %{?buildroot}/%{daoshome}
+mv %{?buildroot}%{_prefix}/{TESTING,lib/%{name}/}
+cp -al ftest.sh src/tests/ftest %{?buildroot}%{daoshome}/TESTING
+find %{?buildroot}%{daoshome}/TESTING/ftest -name \*.py[co] -print0 | xargs -r0 rm -f
+#ln %{?buildroot}%{daoshome}/{TESTING/.build_vars,.build_vars-Linux}.sh
+mkdir -p %{?buildroot}%{daoshome}/utils/py
+cp -al src/utils/py/daos_api.py %{?buildroot}%{daoshome}/utils/py
+cp -al src/utils/py/daos_cref.py %{?buildroot}%{daoshome}/utils/py
+cp -al src/utils/py/conversion.py %{?buildroot}%{daoshome}/utils/py
+mkdir -p %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/
+echo "%{_libdir}/daos_srv" > %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/daos.conf
+
+%post server -p /sbin/ldconfig
+%postun server -p /sbin/ldconfig
+
```

Comment:
presumably at some point we want uninstall scriptlet?

Reply:
Indeed.  As details about what sorts of things should be done as a daos _uninstall_ task emerge, they can be added to this `%postun`.